src/test/app/AMM_test.cpp

```diff
@@ -2986,6 +2985,208 @@ struct AMM_test : public jtx::AMMTest
+        // payment and withdrawl transactions, against an XRP/IOU AMM pool
+        for (STAmount const& money :
+             {STAmount(USD, 1, -3), STAmount(USD, 1, 0), STAmount(USD, 1, 6)})
+        {
+            testAMM([&](AMM& ammAlice, Env& env) {
+                fund(env, gw, {bob}, XRP(1'000), {USD(10'000'000)}, Fund::Acct);
+                env.close();
+
+                // Note: The correct-ness of the balances post-payment,
+                // deposit and withdraw transactions has not been validated.
+                // This has been already validated elsewhere in other unit tests
+                // Rather, the AMM Invariant condition is checked after every
+                // relevant transaction.
+
+                // Note: Despite depositing USD (which is the second asset in
+                // the AMM pool), unit tests use a convention of asset1In to
+                // denote single-asset deposits. It is a similar case with the
+                // AMMWithdraw transaction.
+                ammAlice.deposit(DepositArg{.account = bob, .asset1In = money});
+
+                BEAST_EXPECT(std::apply(
+                    [&](STAmount pool1, STAmount pool2, STAmount lpToken) {
+                        return pool1 * pool2 >= (lpToken * lpToken);
+                    },
+                    ammAlice.balances(XRP, USD)));
+
+                // payment transaction: sendmax is the parameterised value
+                env(pay(bob, carol, STAmount{XRP, 1, 7}),
+                    sendmax(money),
+                    txflags(tfPartialPayment));
+                env.close();
+
+                BEAST_EXPECT(std::apply(
+                    [&](STAmount pool1, STAmount pool2, STAmount lpToken) {
+                        return pool1 * pool2 >= (lpToken * lpToken);
+                    },
+                    ammAlice.balances(XRP, USD)));
+
+                // withdraw transaction
+                ammAlice.withdraw(
+                    WithdrawArg{.account = bob, .asset1Out = money});
+
+                std::cout << "AMM Balances after the withdraw txn: "
+                          << std::get<0>(ammAlice.balances(XRP, USD)) << " "
+                          << std::get<1>(ammAlice.balances(XRP, USD)) << " "
+                          << std::get<2>(ammAlice.balances(XRP, USD))
+                          << std::endl;
+
+                BEAST_EXPECT(std::apply(
+                    [&](STAmount pool1, STAmount pool2, STAmount lpToken) {
+                        return pool1 * pool2 >= (lpToken * lpToken);
+                    },
+                    ammAlice.balances(XRP, USD)));
+            });
+        }
+
+        // test AMM with tiny pool sizes and tiny payment amounts, XRP/IOU
+        // Initial AMM Pool composition (1 drop XRP, 1e-6 USD), 0 trading fees
+        // Swap transaction: Amount = 1e-7 USD, sendmax = 1 drop XRP
+        // Final expected pool composition: 2 drop XRP, 9e-7 USD
+        testAMM(
```

Comment:
I believe the two below unit tests are still useful. They validate small payment transactions against a small AMM pool. But if you feel otherwise (or if there are similar tests), I can remove them.